### ISSUE TITLE
[EventHub] ErrorCodes.UnknownError: Connection in an unexpected error state.

### ISSUE THREAD

**Issue by v-xuto** [Azure/azure-sdk-for-python#22961]
We are running live Tests against other clouds like US Gov and Azure China Cloud. The goal is to check whether new azure sdk package work with other clouds or not.

**Error Description:**
When running the test [test_send.py::test_send_with_partition_key](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/eventhub/azure-eventhub/tests/livetest/synctests/test_send.py#L25) and its [async test](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1348591&view=logs&j=9fc8d1f9-033d-56cc-2df9-e10e0c8a865b&t=557cb5b7-0334-554f-6123-3e8c612f7381&s=2433c88c-0bee-5b70-d3e8-94864d62412b) in `windows2019_36` on `China` cloud, it runs failed and the error message is shown as following, for more details please check [here](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1348591&view=logs&j=9fc8d1f9-033d-56cc-2df9-e10e0c8a865b&t=557cb5b7-0334-554f-6123-3e8c612f7381&s=2433c88c-0bee-5b70-d3e8-94864d62412b):
![image](https://user-images.githubusercontent.com/57166602/152755288-361eaf1f-a1b7-45b9-af89-a24d1cc35a14.png)

**Expected Behavior:**
Test `test_send.py::test_send_with_partition_key` passed in `windows2019_36` on `China` cloud.
In the local test, the probability of passing is `50%`.

@benbp , @jameszliao-msft , @lmazuel , @lilyjma , @ramya-rao-a and @annatisch for notification.

**Comment by swathipil** [comment]
Hi @v-xuto - I think this may take a little more digging into. However, @yunhaoling mentioned that this error may be happening due to poor connection because of the long distance from US to China. Can you try increasing the `timeout` value to `10000` (or something greater) and seeing if that improves test passing rate?

**Comment by v-xuto** [comment]
@swathipil I have updated `timeout` value to 10000. But its result is still `Fail`. For more details in pipeline please check [here](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1374219&view=results).

**Comment by swathipil** [comment]
@benbp - That would be very helpful. Do you know where/how we would set the agent VM location to test this?

**Comment by benbp** [comment]
We would need to spin up a separate agent pool and then configure the test matrix entries to reference that pool. I'll start a conversation with you and @mikeharder.

**Comment by lmazuel** [comment]
If you can't have it in China (different permissions level from the US), there are many location in south korea, india, east asia,that might be able to simplify your scenario (just saying if it helps)

**Comment by mikeharder** [comment]
Our SDK and tests should be reliable even if the client and service are in distant regions.  If we are seeing test failures in distant regions, to me this seems like a bug in either the SDK or the tests that should be investigated and fixed.  Moving the client to a closer region might hide a real issue that needs to be fixed.

**Comment by swathipil** [comment]
Thanks @mikeharder! Will continue digging into the tests to see if there's a best way to fix this.

**Comment by swathipil** [comment]
Hey @v-xuto - sorry for the long wait. It looks like the builds that you linked were deleted. Would you be able to post your error log here?

**Comment by v-xuto** [comment]
@swathipil This is the latest log：https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1397767&view=results

**Comment by swathipil** [comment]
@v-xuto - I've confirmed that Windows is much slower than the other platforms, which would explain why this fails on only Windows. Based on this, I'm still thinking this is an issue with the timeout being set too low. Would you be able to try setting timeout to 20000, 30000, 40000, 50000, 60000, etc. and see if there's a change in rate of tests passing?

**Comment by mikeharder** [comment]
@swathipil: How much slower is Windows, and what is the scenario?  Perf should be similar across Windows and Linux for most scenarios.

**Comment by swathipil** [comment]
@mikeharder - just got clarification, and it looks like I misunderstood. Windows is slower for file operations, but I misconstrued that this affected the tests somehow (which it shouldn't). Thanks for the correction!

**Comment by v-xuto** [comment]
@swathipil By increasing the timeout time, the pass rate does not improve. This makes me think whether the test has much to do with timeout. 
I also tried to debug the test, waiting 1~2 minutes when running to this line of code 
`received = partition.receive_message_batch(timeout=5000)`, but the test still failed.

**Comment by swathipil** [comment]
@v-xuto Thanks for testing this out! I was actually able to reproduce this error locally when testing against an EH namespace where the location is set far away and has a long round trip (specifically, Australia). I'm currently debugging right now to see if we're retrying incorrectly or whether there's a problem elsewhere. Will have more updates to you in the next week.

**Comment by swathipil** [comment]
Hey @v-xuto! Just wanted to give you an update: I believe that the issue had to do with our retry logic in our test. I have a PR out to fix this and am working through a few comments on the PR. Hoping to have this merged by next week.  Thanks @mikeharder for your suggestions on retry and reproducing with location set to Australia!

More details: We were using the underlying `uamqp` ReceiveClient to receive in the test (which doesn't have the retry logic), rather than the EH Consumer Client (which has the retry logic). So, this should only be an issue in our test and not an issue with our SDK. Once I added the retry logic to the test, I was no longer getting this error.

**Comment by swathipil** [comment]
Hey @v-xuto! I just merged the potential fix for this issue. (Sorry about the long wait time on this!) Would you be able to re-run this test on the China cloud and let me know if the test passes? Thanks!

**Comment by v-xuto** [comment]
@swathipil   I have renew a eventhub weekly pipeline, but it still failed in China cloud.
Error: `TimeoutError: Authorization timeout.`
Failed test: `test_send.py::test_send_with_partition_key`, `test_send_async.py::test_send_with_partition_key_async`
More details click here: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1498072&view=results

**Comment by swathipil** [comment]
@v-xuto Would you be able to run against this PR? https://github.com/Azure/azure-sdk-for-python/pull/24014

Updated timeout here and am retrying when there's a timeout error.

**Comment by v-xuto** [comment]
@swathipil I have renew a eventhub weekly pipeline by your PR https://github.com/Azure/azure-sdk-for-python/pull/24014. But it only runs on the `Public` Cloud. 
Pipeline results: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1506942&view=results

If you want test weekly pipeline by your PR, you need to refer to PR https://github.com/Azure/azure-sdk-for-python/pull/21715 and make changes in your PR https://github.com/Azure/azure-sdk-for-python/pull/24014. accordingly.

**Comment by v-xuto** [comment]
@swathipil Is there any progress on the fix of this issue? If there is any progress, please keep us informed. Thanks a lot.

**Comment by swathipil** [comment]
Hey @v-xuto - Really sorry about the delay! I had to prioritize a few other things, unfortunately. However, I was able to reproduce this error that you were running into on a Linux VM. Strangely enough, I know that before I made an update to this test, it was passing on Linux but not Windows. I need to dig a little more to find the source of the error, before I can fix this.

Thanks for checking in! This is definitely on my TODO list, and I will keep you updated on any progress.
